### PR TITLE
added restricted days into redux

### PR DESCRIPTION
**Pull request by aldy97**
# Changes
- Selected days on which users can not go to school now can be added into redux 
- When Restriction is mounted the inner state is initialized to be the state stored in redux

## Motivation and Context
- Saved user's days selected so that it can be used to generate schedules that work for them timewise
- Restriction should be able to memorize user's input 

## How Has This Been Tested?
- Days can be selected and unselected freely, and the change can be reflected in the redux dev tool
- Restrction's rendering and its inner state is always sync with the corresponding state in redux 

# Related Issues
- The way days selected stored in redux is in array of numbers, and they are mapped one-to-one from zero-index
- For example, in redux, days: [0, 1, 3], means the user has claimed that he/she cant go to school on Monday, Tuesday and Thursday

### CODE REVIEW

File: client/src/actions/HomeActions.ts [hunk 1]
```diff
@@ -25,4 +26,14 @@ export interface AddCourseSections {
   sections: CourseObjectProps[];
 }
 
-export type HomeActions = Switch | AddCourse | AddCourseSections | SelectTerm;
+export interface SelectDays {
+  type: typeof SELECTDAYS;
+  days: number[];
```
Comment (johnagl) [inline client/src/actions/HomeActions.ts:35]: Representing days as an array of numbers without defining what the numbers mean is a bit confusing. Let's not use magic numbers here - unless we're defining what they mean somewhere else. We could define an enum such as:
{
 Monday: 0 
 Tuesday: 1
 etc.
}

If a developer looks at the redux store and sees days [0, 1, 2, 3], they will have no clue what that means XD.

**Review by yehee — APPROVED** [review]

**Comment by yehee** [comment]
@aldy97 Uhh accidentally hit the approve button, please disregard it 😢

**Comment by johnagl** [comment]
Okay, so I took a look at this branch and realized we could finish implementing day restrictions with it. So I ended up fixing merge conflicts, making a few minor tweaks to the scheduler algorithm, and got the day restrictions to work. It was mainly combining @aldy97's work with @resurreccionl's scheduler work.

In order to test, add courses/restrictions. Only schedules with none of the day restrictions should appear. 

Good courses to test are BIOL 112 and CHEM 121 - both term 1, since both of these courses have lecture sections on MWF and TTH so you can more easily visualize which sections are being filtered out.

@ubclaunchpad/life-at-ubc if someone could double check this, would be greatly appreciated :D. 

Related issue #127

File: client/src/__tests__/unit/Restriction.test.tsx [hunk 2]
```diff
@@ -4,24 +4,28 @@ import Restrictions, { weekDays } from "../../components/Restrictions";
 import { findTestWrapper } from "../../util/findTestWrapper";
 
 describe("Restrictions", () => {
-  const wrapper = mount(<Restrictions />);
-  const buttons = findTestWrapper(wrapper, "button");
+  // const wrapper = mount(<Restrictions />);
```
Comment (yehee) [inline client/src/__tests__/unit/Restriction.test.tsx]: Do we want to clean up commented lines of code?

File: client/src/components/Restrictions.tsx [hunk 3]
```diff
@@ -1,9 +1,15 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
+// import TimePicker from "./TimePicker";
```
Comment (yehee) [inline client/src/components/Restrictions.tsx]: Not using it anymore?

File: client/src/components/Restrictions.tsx [hunk 4]
```diff
@@ -14,11 +20,11 @@ const ButtonGroup = styled.div`
   justify-content: center;
 `;
 
-const TimePickerGroup = styled.div`
-  display: flex;
-  flex-wrap: wrap;
-  justify-content: center;
-`;
+// const TimePickerGroup = styled.div`
```
Comment (yehee) [inline client/src/components/Restrictions.tsx]: ^

File: client/src/components/Restrictions.tsx [hunk 5]
```diff
@@ -63,13 +78,31 @@ function Restrictions() {
           );
         })}
       </ButtonGroup>
-      <Title title="What time?"></Title>
+      {/* <Title title='What time?'></Title>
```
Comment (yehee) [inline client/src/components/Restrictions.tsx]: ^

**Review by yehee — APPROVED** [review]
Looks good to me! Nothing stands out to me, just a few lines to be cleaned up and it should be good to go!

Edit: also tested a few cases locally with CPSC 416, 415, 319, 310, etc.

File: client/src/components/Restrictions.tsx [hunk 6]
```diff
@@ -41,13 +55,14 @@ function Restrictions() {
...
+
   return (
     <Section>
-      <Title title="3. Add Restrictions" data-test="title"></Title>
-      <Title
-        title="What days do you want to go to school?"
-        data-test="title"
-      ></Title>
+      <Title title="3. Add Restrictions"></Title>
+      <Title title="What days can you not go to school?"></Title>
```
Comment (jacksonyuanjx) [inline client/src/components/Restrictions.tsx]: Would it make sense to change this to "What days do you not want to go to school?"?

**Review by jacksonyuanjx — APPROVED** [review]

File: client/src/components/Restrictions.tsx (hunk 6 above)
Comment (jacksonyuanjx) [reply client/src/components/Restrictions.tsx]: other than that, everything LGTM!

**Comment by aldy97** [comment]
wait I dont even remember what I wrote lol

File: client/src/__tests__/unit/Restriction.test.tsx (hunk 2 above)
Comment (johnagl) [reply client/src/__tests__/unit/Restriction.test.tsx]: Yeah we do, I was a bit lazy, there was a bunch of already commented out code from before :D.

File: client/src/components/Restrictions.tsx (hunk 3 above)
Comment (johnagl) [reply client/src/components/Restrictions.tsx]: I'll remove these..

File: client/src/components/Restrictions.tsx (hunk 4 above)
Comment (johnagl) [reply client/src/components/Restrictions.tsx]: And these...

File: client/src/components/Restrictions.tsx (hunk 6 above)
Comment (johnagl) [reply client/src/components/Restrictions.tsx]: Hmm yeah, this sounds better, I'll make those changes tmr morning and push so people can pull the latest.

File: client/src/__tests__/unit/Restriction.test.tsx (hunk 2 above)
Comment (yehee) [reply client/src/__tests__/unit/Restriction.test.tsx]: Yeah I figured haha. Thanks for taking on this 🙏

**Pull request merged**